intake/source/base.py

```diff
@@ -215,6 +222,27 @@ def to_spark(self):
         """
         raise NotImplementedError
 
+    @property
+    def entry(self):
+        if self._entry is None:
+            raise ValueError("Source was not made by a catalog")
+        if self._entry() is None:
+            raise RuntimeError("Tried to recover entry, but it was gone")
+        return self._entry()
+
+    def clone(self, **kwargs):
+        return self.entry(**kwargs)
```

Comment:
should this return `self.entry(**kwargs)()` so we get a `Datasource` back?

Reply:
"calling" the entry makes a data source :)
I know, I know....

Reply:
Sorry, I had lost track of the layer here :sheep: